connect_ferc1_to_eia.py

```diff
@@ -42,497 +42,785 @@
-                   'plant_id_ferc1',
-                   'capacity_factor',
-                   'capacity_mw',
-                   'net_generation_mwh',
-                   'opex_fuel',
-                   'opex_fuel_per_mwh',
-                   'fuel_cost',
-                   'fuel_mmbtu',
-                   'construction_year',
-                   'installation_year',
-                   'primary_fuel_by_mmbtu',
-                   'plant_type',
-                   'record_id',
-                   # we don't need this for the record linkage, but we do want
-                   # it when we connect depreciation to ferc.
-                   'opex_production_total',
-                   ]
-    fpb_cols_to_use = ['report_year',
-                       'utility_id_ferc1',
-                       'plant_name_ferc1',
-                       'utility_id_pudl',
-                       'fuel_cost',
-                       'fuel_mmbtu',
-                       'primary_fuel_by_mmbtu']
-
-    steam = (
-        pudl_out.plants_steam_ferc1().
-        merge(
-            pudl_out.fbp_ferc1()[fpb_cols_to_use],
-            on=['report_year',
-                'utility_id_ferc1',
-                'utility_id_pudl',
-                'plant_name_ferc1'
-                ],
-            how='left')[cols_to_use]
-        .pipe(pudl.helpers.convert_cols_dtypes,
-              'ferc1', 'ferc1 plant records').
-        # dropna().
-        rename(columns={
-            'fuel_cost': 'total_fuel_cost',
-            'fuel_mmbtu': 'total_mmbtu',
-            'opex_fuel_per_mwh': 'fuel_cost_per_mwh',
-            'primary_fuel_by_mmbtu': 'fuel_type_code_pudl',
-            'record_id': 'record_id_ferc1', }).
-        set_index('record_id_ferc1').
-        assign(
-            fuel_cost_per_mmbtu=lambda x: x.total_fuel_cost / x.total_mmbtu,
-            heat_rate_mmbtu_mwh=lambda x: x.total_mmbtu / x.net_generation_mwh,
-            plant_id_report_year=lambda x: x.plant_id_pudl.map(
-                str) + "_" + x.report_year.map(str)
+        return self.train_df
+
+    def prep_ferc_data(self):
+        """
+        Prepare FERC1 steam data for record linkage with EIA master unit list.
+
+        This method grabs two tables from `pudl_out` (`plants_steam_ferc1`
+        and `fuel_by_plant_ferc1`) and ensures that the columns the same as
+        their EIA counterparts, because the output of this method will be used
+        to link FERC and EIA.
+
+        This method relies on:
+        * pudl_out: an instance of `pudl.output.pudltabl.PudlTabl()`
+
+        Returns:
+            pandas.DataFrame: a cleaned table of FERC1 steam plant records with
+            fuel cost data.
+
+        """
+        fpb_cols_to_use = [
+            'report_year', 'utility_id_ferc1', 'plant_name_ferc1',
+            'utility_id_pudl', 'fuel_cost', 'fuel_mmbtu',
+            'primary_fuel_by_mmbtu']
+
+        logger.info("Preparing the FERC1 steam table.")
```

Comment:
This is consistent for Catalyst, I think it'd be better if "doing a thing" was a debug level message, and info got used for actual results.

debug:  "Preparing the FERC1 steam table." (Unless I'm scared that the code isn't running, who cares? Debug.)
info: "FERC1 steam table identified 215 records from CA." (Cool! The whole reason I ran this thing was to find out about CA, thanks for the INFO.)